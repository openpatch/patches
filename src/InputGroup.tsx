import { ReactElement } from "react";
import { css } from "@emotion/react";
import { Box } from "./Box";
import { InputAddon } from "./InputAddon";
import { InputProps } from "./Input";

export type InputGroupProps = {
  children?: ReactElement<InputProps>;
  addonLeft?: string;
  addonRight?: string;
};

export const InputGroup = ({
  children,
  addonLeft,
  addonRight,
}: InputGroupProps) => {
  return (
    <Box
      display="flex"
      boxShadow="small"
      borderRadius="small"
      borderColor="neutral.200"
      borderStyle="solid"
      borderWidth="standard"
      css={css`
        & > input {
          border: none;
          border-radius: 0;
          flex: 1;
        }
      `}
    >
      {addonLeft && (
        <InputAddon
          css={(theme) => css`
            border-right-color: ${theme.colors.neutral["200"]};
            border-right-width: ${theme.borderWidths.standard};
            border-right-style: solid;
            border-bottom-left-radius: ${theme.radii.small};
            border-top-left-radius: ${theme.radii.small};
          `}
        >
          {addonLeft}
        </InputAddon>
      )}
      {children}
      {addonRight && (
        <InputAddon
          css={(theme) => css`
            border-left-color: ${theme.colors.neutral["200"]};
            border-left-width: ${theme.borderWidths.standard};
            border-left-style: solid;
            border-bottom-right-radius: ${theme.radii.small};
            border-top-right-radius: ${theme.radii.small};
          `}
        >
          {addonRight}
        </InputAddon>
      )}
    </Box>
  );
};
