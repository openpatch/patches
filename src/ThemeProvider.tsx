import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
