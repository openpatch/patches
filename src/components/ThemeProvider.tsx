import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
