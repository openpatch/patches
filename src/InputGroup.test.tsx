
import { render } from "@testing-library/react";
import { InputGroup, InputGroupProps } from "./InputGroup";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: InputGroupProps) =>
render(
    <ThemeProvider>
        <InputGroup {...props} />
    </ThemeProvider>
);

test("should render", () => {
   renderWithTheme();
});

test("should match snapshot", () => {
    const { container } = renderWithTheme();
    expect(container).toMatchSnapshot();
});