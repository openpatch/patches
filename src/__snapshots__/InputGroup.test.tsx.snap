// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.05);
  border-radius: 4px;
  border-color: #CBD2D9;
  border-style: solid;
  border-width: 1px;
}

.emotion-0>input {
  border: none;
  border-radius: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0 emotion-1"
      display="flex"
    />
  </div>
</div>
`;
